charms/benchmark/__init__.py

```diff
@@ -118,9 +123,19 @@ def start():
         If the cabs-collector charm is installed, take a snapshot
         of the current profile data.
         """
+        # Do profile data collection immediately on this unit
         if os.path.exists(COLLECT_PROFILE_DATA):
             subprocess.check_output([COLLECT_PROFILE_DATA])
 
+        # Tell collector charm the action_id via the benchmark relation.
+        # The collector will pass the action_id to its peers,
+        # triggering profile data collection on all collectors in the
+        # environment.
+        for rid in relation_ids('benchmark'):
```

Comment:
There's actually a slight problem with this. The relation name isn't always going to be benchmark so it's hard to say what this relation_id will be.